agent/test-agent-cli/design/RUNBOOK.md

```diff
@@ -0,0 +1,118 @@
+test-agent-cli get-secret secret-key
+
+# To send any error encountered in test
+test-agent-cli send-error error-message
+
+# To create the Tar file of the results, Marking task_state as completed and handling keep_running behaviour
+# Send the path and name of results directory where all the results have been saved
+test-agent-cli terminate --results-dir results_directory
+```
+2. Create a docker file(to provide all the required resources to run the Bash script along with the Command Line Interface in a single pod), that will be used to create image that will run on the Test Pod.(See below example)
+In this docker file make sure to mark entry point as the Test Bash script created in Step 1. Add the commands to install the packages required for Bashscript in docker file. 
+
+```shell
+# syntax=docker/dockerfile:1.1.3-experimental
+
+ARG BUILDER_IMAGE
+FROM ${BUILDER_IMAGE} as build
+
+ARG ARCH
+USER root
+
+# We need these environment variables set for building the `openssl-sys` crate
+
+ENV PKG_CONFIG_PATH=/${ARCH}-bottlerocket-linux-musl/sys-root/usr/lib/pkgconfig
+ENV PKG_CONFIG_ALLOW_CROSS=1
+ENV CARGO_HOME=/src/.cargo
+ENV OPENSSL_STATIC=true
+ENV CARGO_HOME=/src/.cargo
+ADD ./ /src/
+WORKDIR /src/agent/test-agent-cli
+RUN --mount=type=cache,mode=0777,target=/src/target 
+cargo install --locked --offline --target ${ARCH}-bottlerocket-linux-musl --path .  --root ./
+
+FROM public.ecr.aws/amazonlinux/amazonlinux:2
+RUN yum update -y 
+&& yum install -y jq 
+&& yum clean all
+COPY --from=build /src/agent/test-agent-cli/bin/test-agent-cli /usr/local/bin/
+COPY --from=build /src/agent/test-agent-cli/examples/example_test_agent_cli/example-test.sh ./
+
+ENTRYPOINT ["/bin/bash", "./example-test.sh"]
+```
+
+3. Create a docker image consists of the Command Line Interface, Bash Script test and all other resources required to run the test(execute following from Bottlerocket Test System  root folder).(Before running this make sure to add the details to create this image using docker build in [makefile](https://github.com/bottlerocket-os/bottlerocket-test-system/blob/develop/Makefile))
+   `make example-test-agent-cli` and tag this image as : `docker image tag example-test-agent-cli example-test-agent-cli:eks`
+4. Create a controller image(execute from Bottlerocket Test System root folder): `make controller`.
+5. Tag these images as eks as: `docker image tag controller controller:eks`
+6. Create any Resource Agent image(if current test depend on any Resource).
+7. Create a kind cluster that will be used for Testsys `kind create cluster --name testsys`.
+8. Load the docker image for Controller, resource and Test on Kind cluster as :
+```shell
+kind load docker-image controller:eks --name cluster-name
+kind load docker-image example-test-agent-cli:eks --name testsys
+```
+8. Create the namespace, Test and Resource CRDs, roles for test and resource as `testsys install --controller-uri controller:eks`. The controller pod created will run the pod for Test:
```

Comment:
We can just say install testsys to the cluster. I will have more in-depth testsys documentation coming.